lib/game.rb

```diff
@@ -13,16 +13,41 @@ def initialize(input:, output:)
   end
 
   def play
+    introduce_the_game
+    play_to_completion
+    bid_farewell
+  end
+
+  private
+
+  attr_reader :grid, :input, :output
+
+  def introduce_the_game
     grid_view.render
+  end
+
+  def play_to_completion
+    until grid.full?
+      take_player_turn
+      take_computer_turn
+    end
+  end
+
+  def take_player_turn
     player_turn.take
     grid_view.render
-    computer_turn.take
-    grid_view.render
   end
 
-  private
+  def take_computer_turn
+    unless grid.full?
```

Comment:
I would make this a guard clause. The thing about guard clauses is that there is less cognitive load. The thing you are worried about is taken care of in the first line. Then you don't have to think about the indentation or what might be returned when the conditional doesn't apply.